source/javascripts/lib/wirsing/filters/switch.coffee

```diff
@@ -0,0 +1,17 @@
+@wirsing ?= {}
+wirsing.filter ?= {}
+
+wirsing.filter.switch =
+  on: (items, key) ->
+    switchObject = {}
+
+    switches = ->
+      if Object.keys(switchObject).length == 0
```

Comment:
Actually I had it pretty much like you suggested. But currently the problem is that I have to recreate the filter in the angular filter everytime the awards get filtered. So it would iterate over all awards to build the `switches` object unnecessarily. But probably this issue doesn't exist anymore once we drop the angular filters.

About naming: Of course it's clear that it's an object when you assign it, but a few lines down it's probably not that clear anymore. In Ruby I would have probably called it `switchHash`, but since in Javascript you call them `Object`s and not `Hash`es, that's why I called it this way. I wouldn't call it `switches`, because – in my opinion – the name suggests that the object is an array.